python/cudf/cudf/core/dataframe.py

```diff
@@ -7444,3 +7688,11 @@ def _reassign_categories(categories, cols, col_idxs):
                 offset=cols[name].offset,
                 size=cols[name].size,
             )
+
+
+def _from_nested_dict(data) -> defaultdict:
+    new_data: defaultdict = defaultdict(dict)
+    for index, s in data.items():
+        for col, v in s.items():
+            new_data[col][index] = v
+    return new_data
```

Comment:
I really don't like this name, since it doesn't say anything either about what it expects as input, nor what it delivers as output. I think what you're doing is transposing a dictionary?

Reply:
Remove it